pennylane/devices/qubit/preprocess.py

```diff
@@ -205,14 +216,22 @@ def validate_measurements(
     If the circuit has an invalid set of measurements, then an error is raised.
 
     Args:
-        circuit (.QuantumTape): the circuit to validate
+        tape (.QuantumTape): the circuit to validate
         execution_config (.ExecutionConfig): execution configuration with configurable
             options for the execution.
+
+    Returns:
+        qnode (pennylane.QNode) or qfunc or tuple[List[.QuantumTape], function]: If a QNode is passed,
```

Comment:
I'm not sure what the return type is from this. `Union[~pennylane.QNode, Callable, tuple[List[.QuantumTape], Callable]]`? If QNode -> QNode and QFunc -> QFunc, I think I'd just state both of those return patterns explicitly.